torch/_functorch/aot_autograd.py

```diff
@@ -2115,6 +2136,31 @@ def compute_overlapping_inputs(fwd_inputs, aliased_input_indices):
                 actual_aliased_indices.add(j_)
     return actual_aliased_indices
 
+
+def _check_can_be_in_graph(mutates_data, mutates_metadata, requires_grad, keep_input_mutations: bool, is_train: bool):
+    if keep_input_mutations and is_train:
```

Comment:
Why not just say `if not keep_input_mutations: return Fales`

Reply:
Can you remind me: why do we want to branch on `is_train` here? It seems to me like even if we have an inference graph, if have an input mutation on an input that requires_grad then we probably shouldn't keep it in the graph.

It occurs to me now that since https://github.com/pytorch/pytorch/pull/111347 landed, it's technically possible for this to happen (we have an input mutation on an input that requires grad, but the graph is an inference graph). But I think this is probably wrong, and we just just keep these input mutations out of the graph anyway.